src/transformers/models/vit_mae/modeling_tf_vit_mae.py

```diff
@@ -1005,13 +1054,47 @@ def build(self, input_shape=None):
                 with tf.name_scope(layer.name):
                     layer.build(None)
 
+    def interpolate_pos_encoding(self, embeddings, height, width) -> tf.Tensor:
+        """
+        This method allows to interpolate the pre-trained position encodings, to be able to use the model on higher
+        resolution images.
+
+        Source:
+        https://github.com/facebookresearch/dino/blob/de9ee3df6cf39fac952ab558447af1fa1365362a/vision_transformer.py#L174
+        """
+
+        batch_size, seq_len, dim = shape_list(embeddings)
+        num_patches = seq_len - 1
+
+        _, num_positions, _ = shape_list(self.decoder_pos_embed)
+        num_positions -= 1
+
+        if num_patches == num_positions and height == width:
+            return self.decoder_pos_embed
+        class_pos_embed = self.decoder_pos_embed[:, :1]
+        patch_pos_embed = self.decoder_pos_embed[:, 1:]
+        h0 = height // self.config.patch_size
+        w0 = width // self.config.patch_size
+        patch_pos_embed = tf.image.resize(
+            images=tf.reshape(
+                patch_pos_embed, shape=(1, int(math.sqrt(num_positions)), int(math.sqrt(num_positions)), dim)
+            ),
+            size=(h0, w0),
+            method="bicubic",
+        )
+
+        patch_pos_embed = tf.reshape(tensor=patch_pos_embed, shape=(1, -1, dim))
+        return tf.concat(values=(class_pos_embed, patch_pos_embed), axis=1)
+
     def call(
         self,
         hidden_states,
         ids_restore,
         output_attentions=False,
         output_hidden_states=False,
         return_dict=True,
+        pixel_values=None,
```

Comment:
We shouldn't need to pass in pixel_values here. From the code below, it looks like we can get the necessary shape from `x` 

```suggestion
```

Reply:
I actually tried using `x` at early stages of doing this, but always ended up with shape mismatch. But when given the **original image dimension**, the dimension mismatch was gone in both the tf and pt versions. 

for a **(3,480,640) image**, we get the below shapes in interpolation function, both at the encoder and decoder parts of the model. so the scale factor `h0` and `w0` depends on `height and width` parameter and `self.config.patch_size`. Since patch size is to be kept constant, then to obtain the required results, we need to provide the same `height` and `width` values that was provided at the encoder part of the model.

![image](https://github.com/huggingface/transformers/assets/121122109/141e2cf2-b8a9-4708-becf-82a23ea4d3df)

This solves the shape mismatch issue and mainly keeps the changes made to the interpolate_pos_encode function as minimum as possible.

Reply:
Yes, the decoder will likely need a different interpolate function from the encoder. The work is in finding out how to take the current position embeddings and scale them to match x

Reply:
```python
def interpolate_pos_encoding(self, embeddings: torch.Tensor) -> torch.Tensor:
        """
        This method is a modified version of the interpolation function for ViT-mae model at the deocder, that 
        allows to interpolate the pre-trained decoder position encodings, to be able to use the model on higher
        resolution images.

        Source:
        https://github.com/facebookresearch/dino/blob/de9ee3df6cf39fac952ab558447af1fa1365362a/vision_transformer.py#L174
        """
        # height and width parameter cannot be found since here the input tensor for decoder (i.e hidden states) is 3d (batch, seq_len, dim) 

        # -1 removes the class dimension since we later append it without interpolation
        embeddings_positions = embeddings.shape[1] - 1
        num_positions = self.decoder_pos_embed.shape[1] - 1

        # Separation of class token and patch tokens
        class_pos_embed = self.decoder_pos_embed[:, 0, :]
        patch_pos_embed = self.decoder_pos_embed[:, 1:, :]

        # To retain the final 3d tensor with the required dimensions
        dim = self.decoder_pos_embed.shape[-1]

        # Increasing a dimension to enable bicubic interpolation
        patch_pos_embed = patch_pos_embed.reshape(1, 1, -1, dim)

        # permute to bring the dimension to be interpolated, to the last 
        patch_pos_embed = patch_pos_embed.permute(0, 3,1, 2)

        # Interpolating the decoder position embeddings wrt embeddings i.e (x).
        # 1 keeps the other dimension constant
        patch_pos_embed = nn.functional.interpolate(
            patch_pos_embed,
            scale_factor=(1,embeddings_positions / num_positions),
            mode="bicubic",
            align_corners=False,
        )

        # Converting back to the original shape
        patch_pos_embed = patch_pos_embed.permute(0,2,3,1).view(1, -1, dim)
        # Adding the class token back
        return torch.cat((class_pos_embed.unsqueeze(0), patch_pos_embed), dim=1)
```
 This was the logic I had, if we could change the interpolation function. It works for my current test cases too. I have explained my logic in the comments. Do correct me if i'm wrong.

Reply:
LGTM!